src/Agent.Worker/Container/ContainerInfo.cs

```diff
@@ -29,32 +31,60 @@ public ContainerInfo(IHostContext hostContext, Pipelines.ContainerResource conta
             this.ContainerCreateOptions = container.Properties.Get<string>("options");
             this.SkipContainerImagePull = container.Properties.Get<bool>("localimage");
             this.ContainerEnvironmentVariables = container.Environment;
+            this.ContainerCommand = container.Properties.Get<string>("command", defaultValue: "");
 
-#if OS_WINDOWS            
+#if OS_WINDOWS
             _pathMappings[hostContext.GetDirectory(WellKnownDirectory.Tools)] = "C:\\__t"; // Tool cache folder may come from ENV, so we need a unique folder to avoid collision
             _pathMappings[hostContext.GetDirectory(WellKnownDirectory.Work)] = "C:\\__w";
             _pathMappings[hostContext.GetDirectory(WellKnownDirectory.Root)] = "C:\\__a";
 #else
             _pathMappings[hostContext.GetDirectory(WellKnownDirectory.Tools)] = "/__t"; // Tool cache folder may come from ENV, so we need a unique folder to avoid collision
             _pathMappings[hostContext.GetDirectory(WellKnownDirectory.Work)] = "/__w";
             _pathMappings[hostContext.GetDirectory(WellKnownDirectory.Root)] = "/__a";
-#endif            
+#endif
+            this.PortMappings.AddRange(ParsePorts(container.Ports));
+            this.MountVolumes.AddRange(ParseVolumes(container.Volumes));
+            this.IsJobContainer = isJobContainer;
+            if (this.IsJobContainer)
+            {
+                this.MountVolumes.Add(new MountVolume("/var/run/docker.sock", "/var/run/docker.sock"));
```

Comment:
windows and linux will have different command for map docker daemon.

Reply:
I just moved this up so that it only gets mapped on non-windows systems. This mirrors existing functionality here: https://github.com/Microsoft/azure-pipelines-agent/blob/master/src/Agent.Worker/Container/DockerCommandManager.cs#L142

```cs
#if OS_WINDOWS
            string dockerArgs = $"--name {displayName} --label {DockerInstanceLabel} {options} {dockerEnvArgs} {dockerMountVolumesArgs} {image} {sleepCommand}";  // add --network={network} and -v '\\.\pipe\docker_engine:\\.\pipe\docker_engine' when they are available (17.09)
#else
            string dockerArgs = $"--name {displayName} --network={network} --label {DockerInstanceLabel} -v /var/run/docker.sock:/var/run/docker.sock {options} {dockerEnvArgs} {dockerMountVolumesArgs} {image} {sleepCommand}";
#endif
```

Reply:
Added